src/NHibernate/Loader/Criteria/CriteriaLoader.cs

```diff
@@ -12,6 +13,28 @@
 
 namespace NHibernate.Loader.Criteria
 {
+	internal static partial class CriteriaLoaderExtensions
+	{
+		/// <summary>
+		/// Loads all loaders results to single typed list
+		/// </summary>
+		internal static List<T> LoadAllToList<T>(this IList<CriteriaLoader> loaders, ISessionImplementor session)
```

Comment:
I can't use `AddRange` here because I need to add `IList` not `IList\<T\>`. And it's List<object> not List<T> underline.

Reply:
`.AddRange(loader.List(this).Cast<T>())` will do it. Anyway the `ArrayHelper.Add(IList, IList)` which is still used in `LoadAllToList` does that casting.

Reply:
Yeah. I meant it won't preallocate. So no difference with AddAll

Reply:
Due to the casting, right, it would need to be written as `.AddRange(loader.List(this).Cast<T>().ToList())` for benefiting of `AddRange`, but then another list would be created in the process.

So now this change looks to me as a workaround optimization for not having the loader generic.

Reply:
Well it's one place that works effectievly. It can be simplified/inlined when generic version is available. I made a look sometime ago - and it's not simple change at all.